### ISSUE TITLE
Software crashing

### ISSUE THREAD

**Issue by akaKarthick** [czbar/ChessForge#635]
Software crashing. 

here is the demonstration video.
https://youtu.be/huDQvvif4DY

**Comment by czbar** [comment]
Another great find. I really appreciate you exploring different scenarios! 

The bug responsible for the crash shows itself in a workbook/chapter with no games. As we can see in your video, first, the scoresheet is not cleared when you switch to the Games view (which is an ominous sign already!) and when, later on, you click on that scoresheet, the program crashes because it tries to access a variation tree (a game) that doesn't exist).  I'll fix it, of course, before I publish version 1.2.

**Comment by akaKarthick** [comment]
https://youtu.be/ykmxHLlSyfM

**Comment by czbar** [comment]
Fixed in private build 1.2.1 at https://chessforge.sourceforge.io/Releases/latest/ChessForge.msi

**Comment by akaKarthick** [comment]
When I try to enter a comment in a non-English language(Tamil) in the intro tab, right next to diagram, the software crashes.

https://youtu.be/t3YQGzfsgSE

**Comment by czbar** [comment]
I tried to repro but it does not crash for me (I used the Google translator and then pasted the text in Tamil):

![image](https://github.com/czbar/ChessForge/assets/7270924/65e2b701-3b4f-4e44-8f6c-fa7e72ac00bc)

When the crash occurs in the video, is that when you try to type in a Tamil character? In my case, I just pasted the text, rather than type, so maybe it only happens when you type.

Can you try and see if pasting also crashes the program in your scenario? (If not, then I'll have to install Tamil on my laptop and try typing.)

**Comment by akaKarthick** [comment]
I've been able to confirm that the bug only occurs when I type in Tamil characters. When I copy and paste Tamil text, the software does not crash.

**Comment by czbar** [comment]
Yes, I have reproduced. It hard crashes the program which I can't even prevent by catching an exception. Super weird and worrisome.

My debug version reports the following system exception  _"**This error may be a bug in the CLR** or in the unsafe or non-verifiable portions of user code. Common sources of this bug include user marshaling errors for COM-interop or PInvoke, which may corrupt the stack"_. 

The crash occurs in a little procedure that I call to process the Windows internal events queue. Basically, that helps to refresh the GUI.  I will remove the call to it in Intro but will leave in other places, for now.

Let me know if you encounter it elsewhere. If so, I may have to completely drop that function.

**Comment by akaKarthick** [comment]
I understand that this is a difficult situation, and I appreciate your patience and cooperation. if i found it elseware i'll report it.

**Comment by czbar** [comment]
I have removed a call to that "little procedure" I mentioned above, and everything seems ok now. If nothing else comes up I will publish 1.3.0 tomorrow or the day after.

Interestingly, I have tried your scenario with the Russian language, as it also uses a non-Latin alphabet (it sits at 0x04XX in Unicode whereas Tamil at 0x0BXX) but it doesn't crash. Something really strange in the .NET internals (specifically the Rich Text Box control).  Ah well, time to move on :)
